### PR TITLE
Implement grammar based on language specification

### PR DESCRIPTION
**Pull request by w31mann**

### CODE REVIEW

**Comment by liamwh** [comment]
Hi @w31mann, thanks for your amazing contribution!

As this repo is in its infancy, I noticed some CI-related issues that weren't your fault! Those issues should now be addressed, so if you sync with main, resolve this conflicts in the grammar, then your changes should be able to be approved 😊

File: queries/highlights.scm [hunk 1]
```diff
@@ -1,76 +1,104 @@
...
-
-; Constants (UPPER_CASE names and Enums)
-((identifier) @constant
-  (#match? @constant "^[A-Z][A-Z0-9_]+$"))
-
-; Functions and Methods (lowercase names followed by parentheses)
-((identifier) @function
-  (#match? @function "^[a-z_][a-zA-Z0-9_]*%("))
+  "borrow"
+] @type
```
Comment (liamwh) [inline queries/highlights.scm:88]: Please ensure you conform to [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md) standards, so this should be @type.builtin

File: test/corpus/packages.txt [hunk 2]
```diff
@@ -2,17 +2,13 @@
 Packages
 ==================
 
-package wasmcloud:hello@1.0.0;
-package wasi:io;
```
Comment (liamwh) [inline test/corpus/packages.txt:6]: This syntax is [still a part of the spec](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#package-names). Please do not remove it from the test cases.

File: queries/injections.scm
Comment (liamwh) [inline queries/injections.scm:1]: Please don't remove the tree sitter language injection for comments.

File: bindings/rust/build.rs [hunk 3]
```diff
@@ -4,16 +4,16 @@ fn main() {
...
     c_config.file(&parser_path);
     println!("cargo:rerun-if-changed={}", parser_path.to_str().unwrap());
 
-    // NOTE: if your language uses an external scanner, uncomment this block:
-    /*
-    let scanner_path = src_dir.join("scanner.c");
-    c_config.file(&scanner_path);
-    println!("cargo:rerun-if-changed={}", scanner_path.to_str().unwrap());
-    */
+    // let scanner_path = src_dir.join("scanner.c");
```
Comment (liamwh) [inline bindings/rust/build.rs:14]: Please delete the commented code if deleting the comment explaining why it is there

File: queries/highlights.scm [hunk 4]
```diff
@@ -1,76 +1,104 @@
...
+(handle
+  (id) @type)
+
+(named_type
+  name: (id) @variable.parameter)
+
+(record_item
+  name: (id) @variable)
+
+(record_field
```
Comment (liamwh) [inline queries/highlights.scm:28]: To comply with the [nvim-treesitter standards](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights), perhaps this should be `@variable.member`?

File: queries/highlights.scm [hunk 5]
```diff
@@ -1,76 +1,104 @@
...
+(enum_items
+  name: (id) @variable)
+
+(enum_body
+  enum_cases: (id) @property)
+
+(resource_method
+  "constructor" @constructor)
+
+[
```
Comment (liamwh) [inline queries/highlights.scm:52]: To comply with [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#keywords) standards, please distinguish between different keyword variants, such as `@keyword.import` (import, export etc.) and `@keyword.type` (record, variant etc)

File: queries/highlights.scm [hunk 6]
```diff
@@ -1,76 +1,104 @@
...
+  "type"
+  "resource"
+  "func"
```
Comment (liamwh) [inline queries/highlights.scm:55]: `@keyword.function` for "func"

File: queries/highlights.scm (hunk 1 above)
Comment (w31mann) [reply queries/highlights.scm]: I was a bit lazy on highlighting but tried to do it better this time.

File: test/corpus/packages.txt (hunk 2 above)
Comment (w31mann) [reply test/corpus/packages.txt:6]: True but you can only have a single package declaration in a source file. Added more, separate tests for package declarations.

File: queries/injections.scm
Comment (w31mann) [reply queries/injections.scm:1]: it's back

File: bindings/rust/build.rs (hunk 3 above)
Comment (w31mann) [reply bindings/rust/build.rs]: I tried a scanner for better comments parsing but was not worth the effort for that simple kind of comments, defined for WIT.
Should be reverted to what it was before. Still some commented code (with explanation) for easy enablement of scanners.

File: queries/highlights.scm (hunk 4 above)
Comment (w31mann) [reply queries/highlights.scm:52]: should  be better now

File: queries/highlights.scm (hunk 5 above)
Comment (w31mann) [reply queries/highlights.scm:88]: should  be better now

File: queries/highlights.scm (hunk 6 above)
Comment (w31mann) [reply queries/highlights.scm]: should  be better now

**Review by liamwh — APPROVED** [review]

**Pull request merged**